src/java/com/puppetlabs/http/client/impl/JavaClient.java

```diff
@@ -268,7 +278,7 @@ private static void executeWithConsumer(final CloseableHttpAsyncClient client,
                 new StreamingAsyncResponseConsumer(new Deliverable<HttpResponse>() {
             @Override
             public void deliver(HttpResponse httpResponse) {
-                futureCallback.completed(httpResponse);
+                futureCallback.completed(httpResponse); // this stops the timer for the request metric
```

Comment:
Yeah, it is confusing. Your commit makes it a bit easier to understand because it has more to do with what's in scope, but I still find it a bit confusing what timer is stopped when. I'll see if I can make that less confusing.